src/Hal/Serializer/CollectionFiltersNormalizer.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Hal\Serializer;
+
+use ApiPlatform\Core\Api\FilterCollection;
+use ApiPlatform\Core\Api\FilterInterface;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Serializer\ContextTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerAwareTrait;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Enhance the result of collection by adding the filters applied on collection.
+ *
+ * @author Samuel ROZE <samuel.roze@gmail.com>
+ */
+final class CollectionFiltersNormalizer implements NormalizerInterface, SerializerAwareInterface
+{
+    use ContextTrait;
+    use SerializerAwareTrait {
+        setSerializer as baseSetSerializer;
+    }
+
+    private $collectionNormalizer;
+    private $resourceMetadataFactory;
+    private $resourceClassResolver;
+    private $filters;
+    private $formats;
+
+    public function __construct(NormalizerInterface $collectionNormalizer, ResourceMetadataFactoryInterface $resourceMetadataFactory, ResourceClassResolverInterface $resourceClassResolver, FilterCollection $filters, array $formats)
+    {
+        $this->collectionNormalizer = $collectionNormalizer;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->resourceClassResolver = $resourceClassResolver;
+        $this->filters = $filters;
+        $this->formats = $formats;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsNormalization($data, $format = null)
+    {
+        return $this->collectionNormalizer->supportsNormalization($data, $format);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = [])
+    {
+        $data = $this->collectionNormalizer->normalize($object, $format, $context);
+        $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null, true);
+
+        if (isset($context['jsonld_sub_level'])) {
```

Comment:
Maybe the Serializer context keys should be renamed?

* `api_has_context`
* `api_resource_class`
* `api_sub_level`